### PR TITLE
Cirrus: Add initial support

### LINKED ISSUE #3: Need to hook up cirrus testing.
- [x] Create new GCE Project
- [x] Grant stake-holders 'owner' IAM role (including James) in new project
- [x] Ask James Russell to move billing account
- [x] Create 'imgts-<number>' IAM service account in new project
      - Use description: "Account for <new repo> access to update VM Image metadata"
      - Create and store key (JSON) file locally for use later
- [x] Grant 'imgts-<number>' IAM roles in new project
      - Service Account User
- [x] Grant 'imgts-<number>' IAM roles in libpod project
      - Compute Image Label Writer
      - Service Account User
- [x] Create 'cirrus-ci' IAM service account in new project
      - Use description: "Account for <new repo> access to manage resources for Cirrus-CI"
      - Create and store key (JSON) file locally for use later
- [x] Grant 'cirrus-ci' IAM roles in new project
      - Storage Object Admin
      - Storage Admin
      - Compute Instance Admin (v1)
      - Compute OS Admin Login
      - Compute OS Login
      - Service Account User
- [x] Grant 'cirrus-ci' IAM roles in libpod project
      - Compute Image User
      - Service Account User
- [x] In Github Org. settings, enable Cirrus-CI App. access to new repo.
- [x] Go to `https://cirrus-ci.com/github/containers/<new repo name>` settings
      - Disable "Require approval for builds from users without write permissions"
      - Enable "Decrypt Secured Environment Variables for builds initialized by:"
      - Use the encryption box to generate `ENCRYPTED[abc123]` database keys for both service accounts (`imgts-<number>` and `cirrus-ci`) JSON file contents.  Note the strings temporarily.
      - Use the encryption box to generate database keys for the libpod project ID and the complete (fake) e-mail name of the `imgts-<number>` service account.
- [x] Add the new google project ID to [the *Google Cloud Platform projects* list under the *Monitored accounts* section of the Stackdriver workspace settings page](https://app.google.stackdriver.com/settings/accounts/cloud_accounts?project=libpod-218412).
- [x] Clone and create working branch for new repo.
- [x] Open PR for working branch
- [x] Port `.cirrus.ci` over, based on another project, like `buildah`.
- [x] Port over `hack/get_ci_vm.sh`, edit as needed until it works.
- [x] Implement simple tasks/scripts, pushing to PR, and updating until Cirrus-CI working properly.
- [x] Finalize PR based on project needs.

### PR DESCRIPTION
**Pull request by cevich**
Fixes #3

### CODE REVIEW

File: contrib/cirrus/test.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+source $(dirname $0)/lib.sh
+
+[[ -x "bin/oci-seccomp-bpf-hook" ]] || \
+    make
+
+echo "Installing oci-seccomp-bpf-hook"
+make install PREFIX=/usr
+
+echo "Executing integration tests"
+make test-integration BATS_OPTS=--tap
```
Comment (vrothberg) [inline contrib/cirrus/test.sh]: Those need to be executed as `root`. Do we need to add a `sudo` here?

Comment (cevich) [reply contrib/cirrus/test.sh]: Nope, we're root by default.  Though the bats test output is not helpful...I'm making a wild-guess the failure has something to do with podman-1.5 and conmon.  Meaning we have ***both*** `/usr/libexec/podman/conmon` *and* `/usr/bin/conmon`.

Comment (cevich) [reply contrib/cirrus/test.sh]: I think we can wait a little for F30 to pick up `podman 1.6.somerthing` and see if that fixes things.

**Comment by cevich** [comment]
note to me: grab new images from https://github.com/containers/libpod/pull/4189 once ready and merged.

**Comment by cevich** [comment]
@rhatdan @vrothberg need some help here: something is happening with podman + --annotation option used by the BATS tests.  On Fedora 30 (fully updated) if I run w/o it there's no problem.  With it, [podman seems to hang](https://cirrus-ci.com/task/6003512571330560?command=main#L39-L45).  I upped the timeout to 10-minutes, but it still hangs (and seems like an excessive value) :confused:  Any idea how to debug what's happening here?

**Comment by vrothberg** [comment]
> @rhatdan @vrothberg need some help here: something is happening with podman + --annotation option used by the BATS tests. On Fedora 30 (fully updated) if I run w/o it there's no problem. With it, [podman seems to hang](https://cirrus-ci.com/task/6003512571330560?command=main#L39-L45). I upped the timeout to 10-minutes, but it still hangs (and seems like an excessive value) confused Any idea how to debug what's happening here?

I can only guess:
Are the images using `crun`? There's some bug in crun that causes it to hang when the hook exits non-zero. I _guess_ that something is causing the hook to fail and then `crun` to hang which in turns lets `podman` hang. Any chance we can get a shell to one of the Cirrus VMs?

**Comment by cevich** [comment]
force-pushed:  temporarily reverted `showrun` helper (added for debugging + timeout support).  Also added printing of the runc/crun package versions.  Just to confirm, I don't believe `crun` is present or being used on F30, but I could be wrong...

**Comment by vrothberg** [comment]
I had another look with a fresh pair of eyes and found a couple of issues in the latest master. I will fix attempt to fix them to push the Cirrus a bit further.

I see a couple of potential deadlocks that Cirrus - for currently unknown reasons - seems to trigger.

**Comment by vrothberg** [comment]
@cevich any chance we could use another VM and/or another image? I debugged it on the machine you gave me yesterday and it looks like the module is not properly compiled. I tracked it down to the `libbcc` code but already burned a considerable amount of time and am scared of another rabbit hole.

I hope that updating the VM and/or image fixes the issue.

**Comment by cevich** [comment]
Testing w/ repo.iovisor.org added to ubuntu setup as in travis.  This appears to provide different versions of bcc libraries and tools.  Perhaps that's causing the problems since there are native packages available.  Perhaps they're conflicting?

**Comment by cevich** [comment]
@vrothberg as discussed - leaving this failing as-is.  I played around a lot more today, but cannot figure this out.  The main difference between Travis and the two Cirrus-CI Ubuntu VMs has to do with the (somehow) "special" bcc libraries/tools found in the `repo.iovisor.org` repository.  I left a comment about this in `setup.sh`.  Unfortunately, even with that repository enabled (un-commented and 'bcc-tools' package added) - some of the tests still fail.

Sorry I couldn't make the tests work :face_with_head_bandage:

**Comment by cevich** [comment]
rebased vs latest master, disabled ubuntu VMs (for now), and force-pushed.

**Comment by cevich** [comment]
Rebased and current build error under F31 is:

```
make[1]: Entering directory '/var/tmp/go/src/github.com/containers/oci-seccomp-bpf-hook/docs'
/usr/bin/go-md2man -in oci-seccomp-bpf-hook.md -out oci-seccomp-bpf-hook.1
make[1]: Leaving directory '/var/tmp/go/src/github.com/containers/oci-seccomp-bpf-hook/docs'
GO111MODULE=on go build -mod=vendor -o bin/oci-seccomp-bpf-hook -ldflags "-X main.version=1.0.0-dev" github.com/containers/oci-seccomp-bpf-hook
# github.com/iovisor/gobpf/bcc
vendor/github.com/iovisor/gobpf/bcc/module.go:98:40: not enough arguments in call to _Cfunc_bpf_module_create_c_from_string
	have (*_Ctype_char, number, **_Ctype_char, _Ctype_int, _Ctype__Bool)
	want (*_Ctype_char, _Ctype_uint, **_Ctype_char, _Ctype_int, _Ctype__Bool, *_Ctype_char)
```

Perhaps should we lock down to a specific commit/version/tag of gobpf?

**Comment by weirdwiz** [comment]
The package builds on F31 if the bcc version is <= 11. Perhaps we could install that version of bcc-devel.

**Comment by cevich** [comment]
Oh okay, ya I can try that.  Thanks for the suggestion...

**Comment by cevich** [comment]
...On F31, the oldest packaged version of bcc I find is `0.12.0`.   @weirdwiz is it possible to update the code here to work with the newer version?  Seems like that needs to happen at some point anyway, if this hook is going to be packaged, no?

**Comment by weirdwiz** [comment]
> ...On F31, the oldest packaged version of bcc I find is 0.12.0. 

You can use bcc-0.10.0-2.fc31.x86_64 on F31, It works from what I tested

> @weirdwiz is it possible to update the code here to work with the newer version? Seems like that 
needs to happen at some point anyway, if this hook is going to be packaged, no?

The incompatibility seems to be because of the gobpf package, I don't think we can do anything on our end at the moment.

**Comment by cevich** [comment]
> You can use bcc-0.10.0-2.fc31.x86_64 on F31, It works from what I tested

Oh, I couldn't find that one for some reason, I stuck in an old F30 0.10.0 version.  I'll drop that F30 repo and see if it works...

> The incompatibility seems to be because of the gobpf package

Okay for now.  Though I would certainly encourage pursuing a fix even at the package-level.  This is central to the core functionality here.  Having it get too far of sync will very quickly become a headache for both contributors and automation-maintainers.

**Comment by rhatdan** [comment]
This is only going to be supported on F31 and greater and eventually RHEL8.  So dropping F30 is fine with me.

**Comment by cevich** [comment]
@rhatdan okay, SGTM.

File: contrib/cirrus/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
+FROM fedora:30
```
Comment (rhatdan) [inline contrib/cirrus/Dockerfile]: Should this be 31?

**Comment by rhatdan** [comment]
Why is the man page missing?

**Comment by cevich** [comment]
> Why is the man page missing?

Yeah...I can't figure that one out either.  I thought it was because `PREFIX` wasn't being set, but...seems like it's not even building it :confused:

File: contrib/cirrus/Dockerfile (hunk 2 above)
Comment (cevich) [reply contrib/cirrus/Dockerfile]: k, fixed.

**Comment by cevich** [comment]
Welp, that's some progress I guess.  We're running F31 now and not failing on the silly man page anymore.  Any clues as to what might be causing the integration tests to die like that?

**Comment by vrothberg** [comment]
> Welp, that's some progress I guess. We're running F31 now and not failing on the silly man page anymore. Any clues as to what might be causing the integration tests to die like that?

`# Podman output: Error: error executing hook `/usr/libexec/oci/hooks.d/oci-seccomp-bpf-hook` (exit code: 1): OCI runtime error`

Can you give me SSH to this machine?

**Comment by cevich** [comment]
> Can you give me SSH to this machine?

Yes...and with this PR comes a `hack/get_ci_vm.sh` as well.

**Comment by vrothberg** [comment]
The build issue comes from a regression in the bcc libs which is fixed in the latest updates of F31.

We also need a newer kernel. The 5.4.8-200.fc31.x86_64  in the VM seems to have regression that surfaced in the hook being unable to load the tracepoints. My F31 workstation now has 5.4.17-200.fc31.x86_64 which runs without issues. I grepped the kernel logs and found a lot of references on bpf between 5.4.8 and 5.4.17 which makes me positive that an updated VM image will fix both, the build issue and the follow up runtime issue we see in the VM.

**Comment by cevich** [comment]
> We also need a newer kernel.

Okay, this is possible, we just need a new VM image.  Since I'm borrowing those from libpod, the new image will have to come from https://github.com/containers/libpod/pull/3901 (until it merges)

**Comment by cevich** [comment]
@vrothberg we seem to be chasing our tails a bit here.  To help stop that cycle, I just made all the tasks 'optional' here by the last commit.  That way this PR can be merged, and y'all can fix things.  When you're ready to have things be required, just revert that disable-commit.

**Comment by cevich** [comment]
Ahh! Look at that, it's mostly green now :grin:

**Comment by rhatdan** [comment]
@cevich ready to merge?  Do we need bors?

**Comment by vrothberg** [comment]
I'll go ahead and merge. Thanks, @cevich !

**Comment by cevich** [comment]
> Do we need bors?

@rhatdan  Bors would be nice to have in concept, and only a few hours work to setup (worst case).  However,   I'm still a bit apprehensive about using it far/wide, since it's so new (to us).  The issue is _removing_ bors will be painful, since humans love their habits.  I'd rather not have to do that on EVERY containers repo if it fails to meet our reliability needs.

Let's let it cook a bit in buildah for, say, 1-3 months or some disastrous calamity, whichever happens first :grinning:

**Comment by rhatdan** [comment]
SGTM
This REPO will not be active enough to matter anyways.

**Pull request merged**